core/src/main/java/org/apache/calcite/plan/hep/HepPlanner.java

```diff
@@ -500,8 +500,7 @@ private void applyRules(
 
   /** Returns whether the vertex is valid. */
   private boolean belongsToDag(HepRelVertex vertex) {
-    Pair<String, List<RelDataType>> key = key(vertex.getCurrentRel());
-    return mapDigestToVertex.get(key) != null;
+    return graph.vertexSet().contains(vertex);
```

Comment:
This method can be removed, we can call this directly in the caller.

Reply:
Agree with hsyuan.